tests/compiles/tstaticlib.nim

```diff
@@ -0,0 +1,19 @@
+import std/[os, osproc, strformat]
+
+
+const dir = "tests/compiles"
+const fileName = dir / "staticlibtest.nim"
+
+block: # bug #18578
+  const libName = dir / "tstaticlib1.a"
+  let (_, status) = execCmdEx(fmt"nim c -o:{libName} --app:staticlib {fileName}")
```

Comment:
```suggestion
  let (_, status) = execCmdEx(fmt"{nim} c -o:{libName} --app:staticlib {fileName}")
```

and add above: `const nim = getCurrentCompilerExe()`

otherwise this breaks `testament --nim:bin/nim_foo`

Reply:
thanks, fixed